### DISCUSSION TITLE
Raspberry Pi Relay Board support

### DISCUSSION THREAD

**Discussion by Salahaldeen-altous** [labgrid-project/labgrid#1174] [General]
Hi Labgrid-team,

I am trying to add support for seedPi relay board. I will use this commit as a reference [d9c24f04](https://github.com/labgrid-project/labgrid/commit/d9c24f04d1c5ecb9e0a85e51a1f762cd81277e6f).

this relay is connected through I2C to raspberry Pi. Do you have any other relays which are connected over I2C interface? 

https://www.seeedstudio.com/Raspberry-Pi-Relay-Board-v1-0.html

Best Regards,
Salahaldeen Altous

**Comment by Emantor** [comment]
Is there a device tree overlay to expose the relay pins as GPIOs? Than the sysfs GPIO integration can be used.

**Reply by Salahaldeen-altous** [reply]
unfortunately they are not a direct GPIO, they are exported over I2C (I2C to GPIO using PCA9535). the good news These relay Board have a [python library](https://github.com/johnwargo/seeed-studio-relay-board) to control. I have tried to use USB-HID-relay as reference but I got an error like this

`$ labgrid-client -p place1 io get`
`labgrid-client: error: unknown resource class SeeedRelay`
`This is likely caused by an error in the environment configuration or invalid`
`resource information provided by the coordinator.`

attached you can find the my changes as a patch file.

Please check the patch file and if this is the way to add a new driver for this seeedrelay I will create a new branch and submit a pull request, but this patch is still not complete.

[Update patch file]
[0001-Add-support-for-Seeed-Studio-Relay-Board-I2C-to-GPIO.zip](https://github.com/labgrid-project/labgrid/files/11421382/0001-Add-support-for-Seeed-Studio-Relay-Board-I2C-to-GPIO.zip)

**Comment by Salahaldeen-altous** [comment]
@Emantor I have notice that I am missed adding SeeedRelay as resource

`@target_factory.reg_resource`
`@attr.s(eq=False)`
`class SeeedRelay(Resource):`
`    """The basic SeeedRelay contains an index and invert`
`   Args:`
`      index (int): index of target relay.`
`     invert  (bool): whether to invert the relay"""`
`    index = attr.ib(default=1, validator=attr.validators.instance_of(int))`
`    invert = attr.ib(default=False, validator=attr.validators.instance_of(bool))`


where is the best place to add this this new resource?
resource/base.py!!
